lib/commands/actions.js

```diff
@@ -1,23 +1,232 @@
+    const [bundleId, dstPath] = await parseContainerPath(remotePath,
+      async (x) => await getAppContainer(device.udid, x));
+    log.info(`Parsed bundle identifier '${bundleId}' from '${remotePath}'. ` +
+             `Will put the data info '${dstPath}'`);
+    if (!await fs.exists(path.dirname(dstPath))) {
+      log.debug(`The destination folder '${path.dirname(dstPath)}' does not exist. Creating...`);
+      await fs.mkdirp(path.dirname(dstPath));
+    }
+    await fs.writeFile(dstPath, new Buffer(base64Data, 'base64').toString('binary'), 'binary');
+    return;
   }
   const dstFolder = await tempDir.tempDir();
-  // It is important to keep the original file name,
-  // so simctl knows where to put the file
   const dstPath = path.resolve(dstFolder, path.basename(remotePath));
   try {
     await fs.writeFile(dstPath, new Buffer(base64Data, 'base64').toString('binary'), 'binary');
-    await addMedia(this.opts.udid, dstPath);
+    await addMedia(device.udid, dstPath);
   } finally {
     await fs.rimraf(dstFolder);
   }
+}
+
+/**
+ * Save the given base64 data chunk as a binary file on the device under test.
+ * ifuse/osxfuse should be installed and configured on the target machine in order
+ * for this function to work properly. Read https://github.com/libimobiledevice/ifuse
+ * and https://github.com/osxfuse/osxfuse/wiki/FAQ for more details.
+ *
+ * @param {Object} device - The device object, which represents the device under test.
+ *                          This object is expected to have the `udid` property containing the
+ *                          valid device ID.
+ * @param {string} remotePath - The remote path on the device. This variable can be prefixed with
+ *                              bundle id, so then the file will be uploaded to the corresponding
+ *                              application container instead of the default media folder, for example
+ *                              '@com.myapp.bla/RelativePathInContainer/111.png'. The '@' character at the
+ *                              beginning of the argument is mandatory in such case.
+ * @param {string} base64Data - Base-64 encoded content of the file to be uploaded.
+ */
+async function pushFileToRealDevice (device, remotePath, base64Data) {
+  await verifyIFusePresence();
+  const mntRoot = await tempDir.tempDir();
+  let isUnmountSuccessful = true;
+  try {
+    let dstPath = path.resolve(mntRoot, remotePath);
+    let ifuseArgs = ['-u', device.udid, mntRoot];
+    if (remotePath.startsWith(CONTAINER_PATH_MARKER)) {
+      const [bundleId, pathInContainer] = await parseContainerPath(remotePath, mntRoot);
+      dstPath = pathInContainer;
+      log.info(`Parsed bundle identifier '${bundleId}' from '${remotePath}'. ` +
+               `Will put the data info '${dstPath}'`);
+      ifuseArgs = ['-u', device.udid, '--container', bundleId, mntRoot];
+    }
+    await mountDevice(device, ifuseArgs);
+    isUnmountSuccessful = false;
```

Comment:
Why not just start with `isUnmountSuccessful` in the `false` state and set it to `true` after unmount.

Reply:
because I still want to remove that temporary folder even if mount fails